### PR TITLE
Tiles - tile page tweaks and changes for 18Chesapeake

### PR DESCRIPTION
**Pull request by michaeljb**
- `/tiles/<tile_name>` and `/tiles/<game_title>/<tile_or_hex_id>` can render multiple tiles, separated by `+`
- `/map` and `/tile` routes now work with trailing slash (e.g., https://www.18xx.games/tiles/8/ )
- adjust city name positioning for 3 and 4 slot cities -- so "Kouchi" and "Kotohira" (and Washington DC) no longer overlap with the city
- reposition private company blockers for 18Chesapeake (fixes hex K3)
- if private company blocker is only part on a tile, center it and render it larger; good for C-P (H2, I3) and B&S (F4, G5) in 18Chesapeake
- fix some hexes on the 18Chesapeake map--location names, revenue, and offboards (some of these changes had been applied before but were effectively reverted with the 18xx-maker import)

Screenshot from 2020-05-14 18-22-01

Screenshot from 2020-05-14 18-33-41

### CODE REVIEW

File: assets/app/view/map_page.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ class MapPage < Snabberb::Component
 
     def render
       game_title = @route.match(ROUTE_FORMAT)[1]
+      game_title = game_title[0..-2] if game_title[-1] == '/'
```
Comment (tobymao) [inline assets/app/view/map_page.rb:15]: can't this be done with regex?

    ROUTE_FORMAT = %r{/map/(.*)/?}.freeze

File: assets/app/view/part/base.rb [hunk 2]
```diff
@@ -113,6 +113,12 @@ def translate
         "translate(#{x} #{y})"
       end
 
+      def scale
+        s = render_location[:scale] || 1
+
+        "scale(#{s})"
```
Comment (tobymao) [inline assets/app/view/part/base.rb:119]: make this a oneline "scale(#{render_location...})"

File: assets/app/view/part/blocker.rb [hunk 3]
```diff
@@ -6,18 +6,37 @@ module View
...
+          ]
+        else
+          [
+            {
+              region_weights: LEFT_CORNER + [13],
+              x: -65,
+              y: 5,
+            },
+            {
+
```
Comment (tobymao) [inline assets/app/view/part/blocker.rb:26]: extra space, maybe make these constants

File: assets/app/view/tiles_page.rb [hunk 4]
```diff
@@ -22,45 +22,49 @@ class TilesPage < Tiles
 
     def render
       dest = @route.match(ROUTE_FORMAT)[1]
+      dest = dest[0..-2] if dest[-1] == '/'
```
Comment (tobymao) [inline assets/app/view/tiles_page.rb:25]: regex

**Comment by kelsin** [comment]
In #393 I added the data file changes (and the files change into json) so will have to solve that conflict but should just be able to remove your game data changes. I added the names to all hexes and offboard revenue and removed those blank hexes.

File: assets/app/view/map_page.rb (hunk 1 above)
Comment (michaeljb) [reply assets/app/view/map_page.rb]: 2b84872

File: assets/app/view/part/base.rb (hunk 2 above)
Comment (michaeljb) [reply assets/app/view/part/base.rb]: 2b84872

File: assets/app/view/part/blocker.rb (hunk 3 above)
Comment (michaeljb) [reply assets/app/view/part/blocker.rb]: 2b84872

File: assets/app/view/tiles_page.rb (hunk 4 above)
Comment (michaeljb) [reply assets/app/view/tiles_page.rb]: 2b84872

**Pull request merged**